### ISSUE TITLE
Pin topic - Returns 403 Forbidden for Global Moderators

### ISSUE THREAD

**Issue by UniversalSuperBox** [NodeBB/NodeBB#9194]
I was asked to report this bug here after a confirmation in https://community.nodebb.org/topic/15272/pin-topic-returns-403-forbidden-for-global-moderators

- **NodeBB version:** v1.16.1
- **NodeBB git hash:** d7044a5e8623ae6b95360ff58e8f6dbc0e90629b
- **NodeJS version:** v12.20.1

- **Installed NodeBB Plugins:** 
```
Active plugins:
	* nodebb-plugin-composer-default@6.5.5 (installed, enabled)
	* nodebb-plugin-dbsearch@4.1.2 (installed, enabled)
	* nodebb-plugin-email-blacklist@1.1.0 (installed, enabled)
	* nodebb-plugin-emailer-mailgun (not installed, enabled)
	* nodebb-plugin-emoji@3.3.0 (installed, enabled)
	* nodebb-plugin-emoji-android@2.0.0 (installed, enabled)
	* nodebb-plugin-emoji-extended@1.1.1 (installed, disabled)
	* nodebb-plugin-emoji-one@2.0.0 (installed, enabled)
	* nodebb-plugin-import (not installed, enabled)
	* nodebb-plugin-import-mybb (not installed, enabled)
	* nodebb-plugin-markdown@8.12.4 (installed, enabled)
	* nodebb-plugin-mentions@2.13.6 (installed, enabled)
	* nodebb-plugin-newsletter (not installed, enabled)
	* nodebb-plugin-question-and-answer@0.8.5 (installed, enabled)
	* nodebb-plugin-recent-cards (not installed, enabled)
	* nodebb-plugin-soundpack-default@1.0.0 (installed, disabled)
	* nodebb-plugin-spam-be-gone@0.7.7 (installed, enabled)
	* nodebb-rewards-essentials@0.1.4 (installed, enabled)
	* nodebb-theme-lavender@5.0.17 (installed, disabled)
	* nodebb-theme-persona@10.3.17 (installed, enabled)
	* nodebb-theme-slick@1.3.7 (installed, disabled)
	* nodebb-theme-vanilla@11.3.10 (installed, disabled)
	* nodebb-widget-essentials@5.0.2 (installed, enabled)
```
- **Database type:** redis
- **Database version:** Redis server v=5.0.7 sha=00000000:0 malloc=jemalloc-5.2.1 bits=64 build=636cde3b5c7a3923
- **Exact steps to cause this issue:**

1. Add a user to the Global Moderators group via "Admin -> Manage -> Admins & Mods" or "Admin -> Manage -> Groups"
2. Try to pin a post in any category as this user

- **What you expected:**

The new topic should be pinned.

- **What happened instead:**

The user receives a `403 Forbidden` error with the following json payload:

```
{
  "status": {
    "code": "forbidden",
    "message": "[[error:no-privileges]]"
  },
  "response": {}
}
```

The user interface reports that the user does not have enough permissions to perform their task.

This issue started occurring in v1.15.5 (or .6, I'm having trouble remembering now...) and was not resolved with our upgrade to v1.16.1.

If I grant the "Moderate" permission to any of the users in the Global Moderators group on any category, they are able to pin topics in the granted category. I can do this from the Privileges page or by adding the user to the "Admins and Moderators" page. However, these users are still unable to pin topics in any other categories despite their Global Moderator membership.

Members of the `administrators` group are able to pin topics in all categories. Members of the Global Moderator group are able to delete, move, and lock topics as expected. It seems the only affected functionality is pinning topics.

I'm running NodeBB behind an nginx reverse proxy, which is proxied again by Cloudflare.

There does not appear to be anything in the `output.log` of the running NodeBB when this error occurs.

**Comment by julianlam** [comment]
Does this only affect topic pins?

What about locks, or topic deletion?

**Comment by UniversalSuperBox** [comment]
Hi @julianlam, thanks so much for the quick responses!

Locks, deletions, and moves do not appear to be affected. My test user in the Global Moderator group can do any of those tasks. I have only found pins to be affected so far.

**Comment by julianlam** [comment]
Thank you :smile:

**Comment by UniversalSuperBox** [comment]
Whoa, nice! My mods will be very happy to hear it's resolved in 1.16.2
